nova/virt/vmwareapi/vmops.py

```diff
@@ -2058,13 +2133,77 @@ def migrate_disk_and_power_off(self, context, instance, dest, flavor):
                                        total_steps=RESIZE_TOTAL_STEPS)
 
         # 1. Power off the instance
-        vm_util.power_off_instance(self._session, instance, vm_ref)
         self._update_instance_progress(context, instance,
                                        step=1,
                                        total_steps=RESIZE_TOTAL_STEPS)
+        vm_util.power_off_instance(self._session, instance, vm_ref)
+
+        # 2. Detach the volumes
+        self._update_instance_progress(context, instance,
+                                       step=2,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+        self._detach_volumes(instance, block_device_info)
+
+        # 3. Detach network interfaces
+        self._update_instance_progress(context, instance,
+                                       step=3,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+        device_change = self._get_network_device_change(
+            vm_ref, instance.image_meta, network_info, operation="remove")
+        vm_util.vm_device_change(self._session, vm_ref, device_change)
```

Comment:
We're still missing some rollback stuff here. Volume should be attached again, if this fails.

Reply:
Added rollback for this.